tests/Graphql/Resolver/ItemResolverFactoryTest.php

```diff
@@ -13,122 +13,87 @@
 
 namespace ApiPlatform\Core\Tests\Graphql\Resolver;
 
-use ApiPlatform\Core\Api\IdentifiersExtractorInterface;
-use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
-use ApiPlatform\Core\DataProvider\SubresourceDataProviderInterface;
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\Exception\ItemNotFoundException;
 use ApiPlatform\Core\Graphql\Resolver\ItemResolverFactory;
 use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
 use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\Dummy;
+use ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\RelatedDummy;
 use GraphQL\Type\Definition\ResolveInfo;
 use PHPUnit\Framework\TestCase;
 use Prophecy\Argument;
 use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
 
 /**
  * @author Alan Poulain <contact@alanpoulain.eu>
+ * @author Kévin Dunglas <dunglas@gmail.com>
  */
 class ItemResolverFactoryTest extends TestCase
 {
     public function testCreateItemResolverNoItem()
     {
-        $mockedItemResolverFactory = $this->mockItemResolverFactory(null, null, [], null);
+        $resolverFactory = $this->createItemResolverFactory(null);
+        $resolver = $resolverFactory(RelatedDummy::class, Dummy::class, 'operationName');
 
-        $resolver = $mockedItemResolverFactory->createItemResolver('resourceClass', 'rootClass');
+        $resolveInfo = new ResolveInfo([]);
+        $resolveInfo->fieldName = 'name';
+        $resolveInfo->fieldNodes = [];
 
-        $resolveInfoProphecy = $this->prophesize(ResolveInfo::class);
-        $resolveInfoProphecy->fieldName = 'rootProperty';
-
-        $this->assertNull($resolver(null, ['id' => 3], null, $resolveInfoProphecy->reveal()));
+        $this->assertNull($resolver(null, ['id' => '/related_dummies/3'], null, $resolveInfo));
```

Comment:
Should I assume that `['id' => 3]` isn't valid anymore (ie: not using IRIs?)

Reply:
Indeed. IRIs are now mandatory (it's valid according to both the GraphQL and the Relay spec).